### DISCUSSION TITLE
when scaling out the minio, it does not work...

### DISCUSSION THREAD

**Discussion by byeongjokim** [milvus-io/milvus#27665] [Q&A and General discussion] [answered]
my milvus storages(4 pvc (500Gi * 4)) are full. so i try to increase minio pods' replicas.

### before milvus yaml
```
...
  dependencies:
    storage:
      inCluster:
        values:
          replicas: 4
```

### after milvus yaml
```
...
  dependencies:
    storage:
      inCluster:
        values:
          replicas: 6
```

two pods are created but they does not work
<img width="621" alt="image" src="https://github.com/milvus-io/milvus/assets/17046036/472fecab-a6e4-4b5d-b5fe-9d0bd7aeeead">

### error messages in my-release-minio-4 pod
```
API: SYSTEM()
Time: 15:11:16 KST 10/12/2023
Error: Marking http://my-release-minio-5.my-release-minio-svc.default.svc.cluster.local:9000/minio/storage/export/v27 temporary offline; caused by Post "http://my-release-minio-5.my-release-minio-svc.default.svc.cluster.local:9000/minio/storage/export/v27/readall?disk-id=&file-path=format.json&volume=.minio.sys": dial tcp 172.24.10.51:9000: connect: connection refused (*fmt.wrapError)
       6: cmd/rest/client.go:124:rest.(*Client).Call()
       5: cmd/storage-rest-client.go:132:cmd.(*storageRESTClient).call()
       4: cmd/storage-rest-client.go:430:cmd.(*storageRESTClient).ReadAll()
       3: cmd/format-erasure.go:415:cmd.loadFormatErasure()
       2: cmd/format-erasure.go:324:cmd.loadFormatErasureAll.func1()
       1: pkg/sync/errgroup/errgroup.go:122:errgroup.(*Group).Go.func1()
ERROR Unable to initialize backend: http://my-release-minio-0.my-release-minio-svc.default.svc.cluster.local:9000/export disk is already being used in another erasure deployment. (Number of disks specified: 6 but the number of disks found in the 1st disk's format.json: 4)
```

I think this is the problem of existed pods(0~3).
So i set replicas 0 then, set replicas 6
But, all pods have error. They cannot connect each other.

How can i scale out the minio pods?
Or
There are other resolutions when my 4 minio-pvc are full without increasing minio pods? (e.g., increase pvc capacity while preserving data)

**Comment by LoveEachDay** [comment, marked as answer]
@byeongjokim To add  more disk capacity to your minio cluster, you can have two methods:
1. Increase the pvc size (depending on your storageclass capacity)
2. add another server pool, try to increase zone (--set minio.zone=2 for milvus-helm)

**Reply by byeongjokim** [reply]
when i increase the pvc size by modifying [this line](https://github.com/milvus-io/milvus-helm/blob/90f620bd95f06c73fb2ff2b8404a12ff95888418/charts/minio/values.yaml#L144), doesn't the existing data disappear?

**Reply by LoveEachDay** [reply]
No, the data are persisted.

**Reply by byeongjokim** [reply]
thanks.

first i will try the second one.

after deploy, i will let you know.

**Reply by byeongjokim** [reply]
@LoveEachDay it works. thanks.
